### PR TITLE
Boeing viz/scrollytelling

### PR DESCRIPTION
**Pull request by dlousa92**
Hey Jacque,

This pull request is for highlighting parts on the plane svg in and out as a user scrolls down through the page. I changed the previous code a bit and I'm now using the **Previous** method provided in the Waypoints documentation here:
http://imakewebthings.com/waypoints/api/previous/

With this I'm fading pieces and labels out as each section is passed. Take a look and let me know what you think! I think I can make the counter code much simpler using the **Next** method as well.

Next, I need to add behaviour for if a user decides to scroll up (😪), but I wanted to submit this so you get an initial look and let me know if you want me to change anything.

### CODE REVIEW

File: assets/_js/custom-viz/boeing/waypoints.js [hunk 1]
```diff
@@ -85,25 +89,50 @@ const Waypoints = () => {
...
+          Array.from(previousNodes).forEach(child => {
+            if (child.classList.contains('part')) {
+              document
+                .getElementById(`${child.getAttribute('data-part')}-label`)
+                .classList.add('js-hidden')
+
+              let previousChild = document.getElementById(
+                `${child.getAttribute('data-part')}-part`
+              )
+              if (previousChild !== null) {
```
Comment (jnschrag) [inline assets/_js/custom-viz/boeing/waypoints.js]: Functionally, this is all fine. However, you might want to consider refactoring this. Generally speaking, having 3 nested `if` statements isn't ideal. Rather, you should look and see where you could do early returns. For example,

```js
if (!previousWaypoint) {
  return
}

// Rest of code here, although it also could be broken up into early returns.
```

File: assets/_js/custom-viz/boeing/waypoints.js [hunk 2]
```diff
@@ -85,25 +89,50 @@ const Waypoints = () => {
...
-    const allParts = Array.from(document.querySelectorAll('.part'))
+    // const allParts = Array.from(document.querySelectorAll('.part'))
     const activeParts = Array.from(paragraph.querySelectorAll('.part'))
+    // const activeSVGs = []
 
     new Waypoint({
       element: paragraph,
       handler: function() {
-        allParts.forEach(function(part) {
+        const previousWaypoint = this.previous()
```
Comment (jnschrag) [inline assets/_js/custom-viz/boeing/waypoints.js]: I would move this variable declaration so it's above the section of code that it's relevant to.

**Comment by dlousa92** [comment]
Thank you for the review! I'll go ahead and start implementing these changes.

**Comment by dlousa92** [comment]
Thank you for all the great feedback Jacque,

I'm really learning a ton working on this project. I implemented these changes as suggested!

**Pull request merged**